contracts/VoteDelegation.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.13;
+import "./interfaces/IVotingEscrow.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract VoteDelegation is Ownable {
+    struct Delegation {
+        address to;
+        uint256 until;
+    }
+
+    mapping(address => Delegation) public delegation;
+    mapping(address => address[]) public delegated;
+
+    uint256 public constant MAX_DELAGATED = 1_000;
+
+    address public veToken;
+
+    event Delegate(
+        address indexed sender,
+        address indexed recipient,
+        uint256 until
+    );
+    event DelegationRemoved(address indexed from, address indexed to);
+    event DelegationDurationIncreased(
+        address indexed from,
+        address indexed to,
+        uint256 until
+    );
+
+    constructor(address _ve) {
+        require(_ve != address(0x0), "_ve 0x0 address");
+        veToken = _ve;
+    }
+
+    /** 
+    @notice get the list of account that have delegated power to _account
+    @param _account account to check
+    @return The list of account addresses.
+    */
+    function getDelegated(address _account)
+        external
+        view
+        returns (address[] memory)
+    {
+        return delegated[_account];
+    }
+
+    /**
+    @notice Delegate voting power to an address.
+    @param _to the address that can use the voting power
+    @param _until delegate until, if under block.timestamp delegation can change immediately
+     */
+    function delegate(address _to, uint256 _until) external {
+        _delegate(_to, _until);
+    }
+
+    /**
+    @notice Delegate voting power to an address.
+    @param _to the address that can use the voting power
+     */
+    function delegate(address _to) external {
+        _delegate(_to, 0);
+    }
+
+    function _delegate(address _to, uint256 _until) internal {
+        Delegation memory existingDelegation = delegation[msg.sender];
+
+        require(
```

Comment:
why can't you change the delegation to another address until after the until period?